From ecf5ba1840d5a83de088f2e3c7114765c3e50c5c Mon Sep 17 00:00:00 2001
From: Theo GOUREAU <theo.goureau-ext@st.com>
Date: Tue, 7 Jan 2025 13:15:25 +0100
Subject: [PATCH 761/841] Revert "feat(stm32mp2): limit low power mode when
 cortex-M0 is running"

With the software workaround on the Cortex-M0+, there is no more need
for low power mode limitation.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git ba15edda27921aed0c33fe01e9e000719e96539e Revert "feat(stm32mp2): limit low power mode when cortex-M0 is running"]
Change-Id: I46618823b15d1f6193a7d9bee7edb2b626677b1f
Signed-off-by: Theo GOUREAU <theo.goureau-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/427331
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/stm32mp2_pm.c | 25 -------------------------
 1 file changed, 25 deletions(-)

diff --git a/plat/st/stm32mp2/stm32mp2_pm.c b/plat/st/stm32mp2/stm32mp2_pm.c
index 7d3a58830..677ad16d6 100644
--- a/plat/st/stm32mp2/stm32mp2_pm.c
+++ b/plat/st/stm32mp2/stm32mp2_pm.c
@@ -469,15 +469,6 @@ bool stm32_pwr_cpu2_state_is_running(uintptr_t pwr_base)
 #endif
 }
 
-bool stm32_pwr_cpu3_state_is_running(uintptr_t pwr_base)
-{
-#if STM32MP21
-	return false;
-#else
-	return (mmio_read_32(pwr_base + PWR_CPU3D3SR) & PWR_CPU3D3SR_CSTATE_MASK) != 0U;
-#endif
-}
-
 static int stm32_pwr_domain_validate_suspend(const psci_power_state_t *target_state)
 {
 	uintptr_t pwr_base = stm32mp_pwr_base();
@@ -501,16 +492,6 @@ static int stm32_pwr_domain_validate_suspend(const psci_power_state_t *target_st
 		}
 	}
 
-	/* If CPU3 is not in reset: limit supported low power modes */
-	if (stm32_pwr_cpu3_state_is_running(pwr_base)) {
-		if (stateid == PWRSTATE_STANDBY ||
-		    stateid == PWRSTATE_LPLV_STOP2 ||
-		    stateid == PWRSTATE_LPLV_STOP1) {
-			WARN("Invalid PSCI power state %x with Cortex M0 running.\n", stateid);
-			return PSCI_E_INVALID_PARAMS;
-		}
-	}
-
 	if (!stm32_freeze_other_core(core_id)) {
 		return PSCI_E_DENIED;
 	}
@@ -1178,12 +1159,6 @@ static void stm32_get_sys_suspend_power_state(psci_power_state_t *req_state)
 		VERBOSE("%s: max_pwr_state = PWRSTATE_LP_STOP2, M33 is running\n", __func__);
 	}
 
-	/* M0 is running: Standby1 and LPLV are not allowed */
-	if (stm32_pwr_cpu3_state_is_running(pwr_base)) {
-		max_pwr_state = PWRSTATE_LP_STOP2;
-		VERBOSE("%s: max_pwr_state = PWRSTATE_LP_STOP2, M0 is running\n", __func__);
-	}
-
 	/* Trace to debug low power mode restriction */
 	if (max_pwr_state != PWRSTATE_STANDBY) {
 		INFO("max_pwr_state=%x C1IMR1=%x C1IMR2=%x C1IMR3=%x CPU2D2SR=%x\n",
-- 
2.47.2

